### PR TITLE
Ashwalker Tribe roundend reporting revamp

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

This PR changes how ashwalkers are listed in the roundend report. Originally, they would be listed as individual "other" antagonists, which isn't a particularly great way of handling it and leads to weirdness like this:

image

_(That's an authentic ninja greentext from a live server, by the by)_

Now, the roundend report gives each ashwalker team their own section within the report, whether or not they succeeded, and some bonus stats.

image

This new roundend report handling also supports having multiple Necropolises (Necropoli?) in the same round. Each tribe has their stats and success reported individually. 

This also slightly modifies the protect_object objective, preventing hard deletes when linked to objects that don't clear their own references.

## Why It's Good For The Game

Gives Ashwalkers a much better roundend report, and handles how the reporting is done in a much more sensible manner.

Reduces how much space that large nests take up with their individual listings.
## Changelog
:cl: Rhials
qol: Ashwalker roundend reporting has been revamped. Glory to the Necropolis!
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Maybe add namings for tribes? So it's also easier to find the tribe you're looking for.

**Comment by Googles-Hands** [comment]
Why change it from Ash Walker to Ashwalker. The imdividual antags are still called "Ash Walker", either change it all or keep it uniform.

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 1]
```diff
@@ -42,3 +38,40 @@
...
+	member_name = "Ash Walker"
+	///A list of "worthy" (meat-bearing) sacrifices made to the Necropolis
+	var/sacrifices_made = 0
+	///A list of how many eggs were created by the Necropolis
+	var/eggs_created = 0
+
+/datum/team/ashwalkers/roundend_report()
+	var/list/report = list()
+
+	report += "<span class='header'>An Ash Walker Tribe inhabited the wastes...</span><br>"
```
Comment (san7890) [inline code/modules/antagonists/ashwalker/ashwalker.dm:53]: do you think it'd be worthwhile to make a `span_header()` for this in this PR? it should definitely be done at some point so i'm wondering if now is a good time or not.

File: code/modules/antagonists/ashwalker/ashwalker.dm [hunk 2]
```diff
@@ -42,3 +38,40 @@
...
+	if(length(members)) //The team is generated alongside the tendril, and it's entirely possible that nobody takes the role.
+		report += "The [member_name]s were:"
+		report += printplayerlist(members)
+
+		var/datum/objective/protect_object/necropolis_objective = locate(/datum/objective/protect_object) in objectives
+
+		if(necropolis_objective)
+			objectives -= necropolis_objective //So we don't count it in the check for other objectives.
+			report += "<b>The [name] was tasked with defending the Necropolis:</b>"
```
Comment (san7890) [inline code/modules/antagonists/ashwalker/ashwalker.dm:62]: these could reasonably be `span_bold()`, pretty sure our HTML ships with class=bold (if it doesn't: wack).

File: code/modules/antagonists/ashwalker/ashwalker.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/antagonists/ashwalker/ashwalker.dm:53]: im killing this whole menu when i get the motivation anyway

File: code/modules/antagonists/ashwalker/ashwalker.dm (hunk 2 above)
Comment (Rhials) [reply code/modules/antagonists/ashwalker/ashwalker.dm:62]: I tried using span_bold(), it didn't seem to do anything.

I concur, this is indeed "wack".

**Comment by AviiCinders** [comment]
The tribe has fallen, billions must die

**Pull request merged**